pytorch_lightning/loops/fit_loop.py

```diff
@@ -174,6 +174,8 @@ def connect(self, epoch_loop: TrainingEpochLoop):
 
     def reset(self) -> None:
         """Resets the internal state of this loop."""
+        if self.restarting:
```

Comment:
Shouldn't it be `not`?

```suggestion
        if not self.restarting:
```

Reply:
Not it is correct. We used to reset in the loop. After we talked, you asked me to move it in the reset functions to make it explicit. Here it is :)